Testing/Unit/PowerShell/Orchestrator/Connect-Tenant.Tests.ps1

```diff
@@ -0,0 +1,98 @@
+$OrchestratorPath = '../../../../PowerShell/ScubaGear/Modules/Orchestrator.psm1'
+Import-Module (Join-Path -Path $PSScriptRoot -ChildPath $OrchestratorPath) -Function 'Connect-Tenant' -Force
+
+InModuleScope Orchestrator {
+    Describe -Tag 'Orchestrator' -Name 'Invoke-Connection' {
+        Context 'When interactively connecting to commercial Endpoints' {
+            BeforeAll {
+                [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSUseDeclaredVarsMoreThanAssignments', 'ConnectParams')]
+                $ConnectParams = @{
+                    LogIn           = $true
+                    M365Environment = "commercial"
+                    BoundParameters = @{}
+                }
+                function Connect-Tenant {}
+                Mock -ModuleName Orchestrator Connect-Tenant -MockWith {@()}
+            }
+            It 'With -ProductNames "aad", connects to Microsoft Graph' {
+                $ConnectParams += @{
+                    ProductNames = 'aad'
+                }
+                $FailedAuthList = Invoke-Connection @ConnectParams
+                $FailedAuthList.Length | Should -Be 0
+            }
+            It 'With -ProductNames "defender", connects to Microsoft Defender for Office 365' {
```

Comment:
No value added calling with each product name.  No logic is exercise on the product names.  Recommeend omitting.

Reply:
I see the logic in not having duplicated code. we would like to print out what product we are currently running the unit test for but we can't use a parameter for the "It" block. This code might have to be duplicated for each product.